File: Source/DotNET/Fundamentals/Geospatial/Point.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Geospatial;

/// <summary>
/// Represents a geographic point with longitude and latitude.
/// </summary>
/// <param name="Longitude">The longitude of the point.</param>
/// <param name="Latitude">The latitude of the point.</param>
public record Point(double Longitude, double Latitude);

File: Source/JavaScript/geospatial/Point.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { field } from '../fieldDecorator';

/**
 * Represents a geographic point with longitude and latitude.
 */
export class Point {
    @field(Number)
    longitude!: number;

    @field(Number)
    latitude!: number;

    /**
     * Creates a new Point instance.
     * @param {number} longitude - The longitude of the point.
     * @param {number} latitude - The latitude of the point.
     */
    constructor(longitude: number, latitude: number) {
        this.longitude = longitude;
        this.latitude = latitude;
    }

    /**
     * Converts the Point to a GeoJSON representation.
     * @returns {object} GeoJSON representation.
     */
    toJSON(): object {
        return {
            type: 'Point',
            coordinates: [this.longitude, this.latitude]
        };
    }
}
